libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerAction.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.AI.QnA
+{
+    /// <summary>
+    /// QnAMaker dialog which uses QnAMaker to get an answer.
+    /// </summary>
+    public class QnAMakerAction : Dialog
+    {
+        private const float DefaultThreshold = 0.3F;
+        private const string DefaultNoAnswer = "No QnAMaker answers found.";
+
+        private QnAMaker qnamaker;
+        private readonly HttpClient httpClient;
+
+        public QnAMakerAction(string knowledgeBaseId, string endpointKey, string hostName, string noAnswer = DefaultNoAnswer, float threshold = DefaultThreshold, Metadata[] strictFilters = null,   HttpClient httpClient = null, [CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+            this.KnowledgeBaseId = knowledgeBaseId ?? throw new ArgumentNullException(nameof(knowledgeBaseId));
+            this.HostName = hostName ?? throw new ArgumentNullException(nameof(HostName));
+            this.EndpointKey = endpointKey ?? throw new ArgumentNullException(nameof(EndpointKey));
```

Comment:
Wrong variable in the nameof operator. Should be : `endpointKey ?? throw new ArgumentNullException(nameof(endpointKey ));`